vector_search/tasks.py

```diff
@@ -148,3 +150,30 @@ def start_embed_resources(self, indexes, skip_content_files):
     # Use self.replace so that code waiting on this task will also wait on the embedding
     #  and finish tasks
     return self.replace(celery.chain(*index_tasks))
+
+
+@app.task(bind=True)
+def embed_new_learning_resources(self, period="daily"):
+    """
+    Embed new resources in period by type
+    """
+    log.info("Running %s embedding task", period)
+    delta = datetime.timedelta(days=1)
+    if period == "weekly":
+        delta = datetime.timedelta(days=7)
+    since = now_in_utc() - delta
```

Comment:
Do you think there's a chance some resources might occasionally fall through the cracks?  For example maybe this task gets occasionally backed up behind a bunch of other tasks and runs late.  Wondering if we should buffer this a bit by making the timedelta something like 25 hours for daily and 7*24+1 hours for weekly?  Or instead of using `now_in_utc() - timedelta`, filter by `created_on__gt=<today's date at a specific time(like 12:00 AM) - 1 day/week>`?

Reply:
I kicked off this task in a django shell `embed_new_learning_resources.delay()`, it frequently overwhelmed my celery workers, though they do start up again, not sure if it will complete successfully, it is still trying.

![Screenshot 2024-12-04 at 11 37 28 AM](https://github.com/user-attachments/assets/8668c76b-db02-488c-88a8-433a6032102f)

Reply:
PS Did not have this problem after switching the new env variables over to openai

Reply:
The error you were seeing could be due to constrained docker resources when running embedding via fastembed  - you might have to increase cpu allocation for docker. I have mine set to the max and it still seems to be overflowing  (but not crashing) during the local embedding process

<img width="505" alt="Screenshot 2024-12-04 at 3 23 47 PM" src="https://github.com/user-attachments/assets/4b3a8eaa-4d8c-4e1c-92c4-478515505c0f">
<img width="414" alt="Screenshot 2024-12-04 at 3 24 44 PM" src="https://github.com/user-attachments/assets/5b6e8df0-d176-4061-aeec-1e2f826e08d8">